## Austin Listings EDA

## Loading the Data Set
- The first step is to load the data set I will be exploring
- This will allow me to analyze and inspect:
    - The shape of my data
    - Null values
    - Columns that may or may not be relevant to my business question

- I also like to output a small portion of the data set to get a quick understanding of how the data looks

In [10]:
import pandas as pd

df = pd.read_csv('../data/listings_Austin.csv')
df.head()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
0,5456,"Walk to 6th, Rainey St and Convention Ctr",8028,Sylvia,NaN,78702,30.26057,-97.73441,Entire home/apt,97.0,2,708,2025-09-02,3.52,1,328,25,NaN
1,6448,"Secluded Studio @ Zilker - King Bed, Bright & ...",14156,Amy,NaN,78704,30.26034,-97.76487,Entire home/apt,160.0,3,339,2025-08-20,1.98,1,316,14,NaN
2,8502,Woodland Studio Lodging,25298,Karen,NaN,78741,30.23466,-97.73682,Entire home/apt,38.0,4,54,2025-05-05,0.28,1,88,1,NaN
3,13035,Historic house in highly walkable East Austin,50793,Margaret Ann,NaN,78702,30.26098,-97.73072,Entire home/apt,145.0,15,19,2025-07-31,0.11,2,321,1,NaN
4,22828,Garage Apartment central SE Austin,56488,David,NaN,78741,30.23614,-97.73225,Entire home/apt,58.0,30,56,2025-08-16,0.30,1,211,3,NaN


In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10533 entries, 0 to 10532
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              10533 non-null  int64  
 1   name                            10533 non-null  str    
 2   host_id                         10533 non-null  int64  
 3   host_name                       10524 non-null  str    
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   10533 non-null  int64  
 6   latitude                        10533 non-null  float64
 7   longitude                       10533 non-null  float64
 8   room_type                       10533 non-null  str    
 9   price                           10517 non-null  float64
 10  minimum_nights                  10533 non-null  int64  
 11  number_of_reviews               10533 non-null  int64  
 12  last_review                     8909 non-nu

## Findings from df.info():
- Next I like to run df.info() to give me a quick, concise summary of very important features of my data set such as shape, data types, and any null values my data may have

**Shape**
- We can see that the shape of this data set is (10533, 18)
- This means there are 10,533 rows/entries and 18 columns/attributes

**Data-types**
- The data set contains 3 distinct data-types
- float64 (decmial values) 
- int64 (whole numbers)
- str (text)

**Null values**
- df.info() shows us the number of rows that dont have NULL values, for each columm
- The following columns have entries whos values are NULL and will be handled later on:
    - host_name             ... 9 NULL values                      
    - neighbourhood_group   ... 10533 NULL values
    - price                 ... 16 NULL values          
    - last_review           ... 1624 NULL values 
    - reviews_per_month     ... 1624 NULL values
    - license               ... 10533 NULL values 


**Other potential issues to investigate**
- 'neighbourhood' column is an integer data type
    - Since we will never perform calculations on a zip code, this column should be a string data type
    - Additionally, during data analysis and visualization, a numerical value that is actually categorical may cause issues with grouping and plotting visuals



In [14]:
df[['price', 'minimum_nights', 'availability_365', 'number_of_reviews']].describe()

,price,minimum_nights,availability_365,number_of_reviews
count,10517.000000,10533.000000,10533.000000,10533.000000
mean,414.537701,7.768157,234.909238,55.858920
std,2896.807897,20.586238,114.102467,104.276866
min,8.000000,1.000000,0.000000,0.000000
25%,86.000000,1.000000,143.000000,2.000000
50%,135.000000,2.000000,263.000000,17.000000
75%,226.000000,3.000000,339.000000,62.000000
max,50000.000000,365.000000,365.000000,1305.000000


## Findings from df.describe()
- I also like to run df.describe() on key numerical columns relevant to my business question
- Running this helps me identify any skewed distributions or outliers that could affect my analysis

**Price:**
- Min: $8 ... suspiciously low, needs investigation
- Max: $50,000 ... likely an outlier that could skew analysis
- Mean: $414 ... likely inflated by extreme values

**Minimum Nights:**
- Max: 365 nights ... a full year minimum stay suggests some listings 
  are not true short term rentals and may need to be filtered out

## Dallas Listings EDA


In [19]:
df = pd.read_csv('../data/listings_Dallas.csv')
df.head()


,id,name,host_id,host_profile_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
0,61878,MODERN LIVING AND FURNISHINGS,300211.0,1.462515e+18,Rita,NaN,District 2,32.81690,-96.82018,Entire home/apt,92.0,30.0,59,2026-03-30,0.31,1.0,292,2,NaN
1,795703,Amazing location walk to Downtown Dallas,4191322.0,1.462534e+18,Michelle,NaN,District 14,32.80327,-96.80976,Entire home/apt,179.0,30.0,71,2024-07-27,0.44,1.0,232,0,NaN
2,1277933,The Santa Fe Suite and Maxfield Parrish Suite,6904484.0,1.462548e+18,C. F. Sandy,NaN,District 11,32.92449,-96.79062,Private room,86.0,1.0,30,2019-11-09,0.27,1.0,298,0,NaN
3,1479800,SMU/Trendy Greenville/with Private Bath,3809024.0,1.462532e+18,Holly,NaN,District 14,32.83103,-96.76380,Private room,150.0,2.0,103,2025-10-26,0.66,3.0,107,1,NaN
4,1507428,"4323 McKinney Ave studio 8, WIFI, Parking",6063232.0,1.462697e+18,Joan,NaN,District 14,32.81961,-96.78996,Entire home/apt,53.0,30.0,75,2026-04-30,0.49,9.0,312,1,NaN


In [20]:
df.info()
df[['price', 'minimum_nights', 'availability_365', 'number_of_reviews']].describe()

<class 'pandas.DataFrame'>
RangeIndex: 6217 entries, 0 to 6216
Data columns (total 19 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              6217 non-null   int64  
 1   name                            6217 non-null   str    
 2   host_id                         6211 non-null   float64
 3   host_profile_id                 6210 non-null   float64
 4   host_name                       6211 non-null   str    
 5   neighbourhood_group             0 non-null      float64
 6   neighbourhood                   6217 non-null   str    
 7   latitude                        6217 non-null   float64
 8   longitude                       6217 non-null   float64
 9   room_type                       6217 non-null   str    
 10  price                           5901 non-null   float64
 11  minimum_nights                  6216 non-null   float64
 12  number_of_reviews               6217 non-null

,price,minimum_nights,availability_365,number_of_reviews
count,5901.000000,6216.000000,6217.000000,6217.000000
mean,328.692594,5.541184,254.163262,43.744571
std,947.322891,18.534695,114.869311,86.823626
min,5.000000,1.000000,0.000000,0.000000
25%,134.000000,1.000000,173.000000,2.000000
50%,204.000000,1.000000,300.000000,14.000000
75%,330.000000,2.000000,350.000000,47.000000
max,50000.000000,365.000000,365.000000,1473.000000


## Dallas Findings Summary

**Shape**
- 6,217 rows and 19 columns
- Fewer listings than Austin (10,533)

**Key Differences from Austin**
- Has an extra column: host_profile_id ... will drop when combining cities
- neighbourhood contains District names, not zip codes like Austin
- host_id and minimum_nights stored as float64 instead of int64
- price has 316 null values vs Austin's 16

**Null Values**
- host_id               ... 6 NULL values
- host_profile_id       ... 7 NULL values
- host_name             ... 6 NULL values
- neighbourhood_group   ... 6217 NULL values 
- price                 ... 316 NULL values 
- last_review           ... 1106 NULL values
- reviews_per_month     ... 1106 NULL values 
- license               ... 6217 NULL values 

**Outliers from df.describe()**
- Price 
    - min: 5 ... very low even for one night 
    - max: 50000 ... same outlier pattern as Austin
- minimum_nights 
    - max: 365 ... minimum stay is one year?

## Fort Worth EDA

In [21]:
df = pd.read_csv('../data/listings_FortWorth.csv')
df.head()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365,number_of_reviews_ltm,license
0,257869,House w/ Pool in Cultural District/close2downtown,1356081,Shayna,NaN,NaN,32.74186,-97.37857,Entire home/apt,NaN,31,377,2024-11-10,2.26,1,0,8,NaN
1,299748,Rosy's Guest House,1544441,Kimberly,NaN,District 9,32.71696,-97.34170,Entire home/apt,NaN,28,361,2025-08-10,2.17,1,0,9,NaN
2,912371,Stay in Ft.Worth Cultural District,4895584,V R,NaN,District 7,32.73643,-97.38305,Entire home/apt,58.0,30,323,2025-07-31,2.11,1,58,3,NaN
3,1885626,Private bedroom in spacious home,7892516,Tibor,NaN,NaN,33.02235,-97.30032,Private room,54.0,1,98,2025-08-25,0.69,1,79,11,NaN
4,2130492,"Charming ""attic room"" by TCU campus",826752,Lee,NaN,District 3,32.71192,-97.36889,Private room,100.0,30,41,2023-06-19,0.31,2,179,0,NaN


In [23]:
df.info()
df[['price', 'minimum_nights', 'availability_365', 'number_of_reviews']].describe()

<class 'pandas.DataFrame'>
RangeIndex: 2081 entries, 0 to 2080
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              2081 non-null   int64  
 1   name                            2081 non-null   str    
 2   host_id                         2081 non-null   int64  
 3   host_name                       2073 non-null   str    
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   1012 non-null   str    
 6   latitude                        2081 non-null   float64
 7   longitude                       2081 non-null   float64
 8   room_type                       2081 non-null   str    
 9   price                           1856 non-null   float64
 10  minimum_nights                  2081 non-null   int64  
 11  number_of_reviews               2081 non-null   int64  
 12  last_review                     1718 non-null

,price,minimum_nights,availability_365,number_of_reviews
count,1856.000000,2081.000000,2081.000000,2081.000000
mean,494.698815,10.316194,235.712158,42.268621
std,4030.294355,23.988418,118.611459,72.323468
min,15.000000,1.000000,0.000000,0.000000
25%,75.000000,1.000000,145.000000,2.000000
50%,113.000000,2.000000,268.000000,12.000000
75%,172.000000,15.000000,344.000000,52.000000
max,50000.000000,365.000000,365.000000,813.000000


## Fort Worth Findings Summary

**Shape**
- 2,081 rows and 18 columns
- Significantly fewer listings than Austin (10,533) and Dallas (6,217)

**Key Differences from Austin and Dallas**
- neighbourhood has 1,069 NULL values — over half the listings are missing a neighbourhood
- No extra columns compared to Austin — same 18 column structure
- price has 225 null values vs Austin's 16 and Dallas's 316

**Null Values**
- host_name              ... 8 NULL values
- neighbourhood_group    ... 2081 NULL values
- neighbourhood          ... 1069 NULL values
- price                  ... 225 NULL values
- last_review            ... 363 NULL values
- reviews_per_month      ... 363 NULL values
- license                ... 2081 NULL values

**Outliers from df.describe()**
- Price
    - min: 15 ... slightly more reasonable than Austin and Dallas
    - max: 50,000 ... same outlier pattern across all three cities
    - mean: $494 ... highest mean of all three cities, heavily skewed by outliers
- minimum_nights
    - max: 365 ... same issue as Austin and Dallas
- neighbourhood
    - 1,069 missing values ... will need to decide whether to drop rows or fill during cleaning